jujugui/static/gui/src/app/components/status/status.js

```diff
@@ -5,23 +5,363 @@
+      return (
+        <tr key={machine.id}>
+          <td>{machine.displayName}</td>
+          <td className={getClass(machine.agent_state)}>
+            {machine.agent_state}
+          </td>
+          <td>{machine.public_address}</td>
+          <td>{machine.instance_id}</td>
+          <td>{machine.series}</td>
+          <td>{machine.agent_state_info}</td>
+        </tr>
+      );
+    });
+    return (
+      <table key="machines">
+        <thead>
+          <tr>
+            <th>Machine</th>
+            <th>State</th>
+            <th>DNS</th>
+            <th>Instance ID</th>
+            <th>Series</th>
+            <th>Message</th>
+          </tr>
+        </thead>
+        <tbody>
+          {rows.sort(byKey)}
+        </tbody>
+      </table>
+    );
+  }
+
+  /**
+    Render the relations fragment of the status.
+    @params {Array} relations The relations as included in the GUI db.
+  */
+  _renderRelations(relations) {
+    const rows = relations.map(relation => {
+      const rel = relation.getAttrs();
+      let name = '';
+      let provides = '';
+      let consumes = '';
+      let scope = '';
+      rel.endpoints.forEach(endpoint => {
+        const application = endpoint[0];
+        const ep = endpoint[1];
+        switch (ep.role) {
+          case 'peer':
+            name = ep.name;
+            provides = application;
+            consumes = application;
+            scope = 'peer';
+            return;
+          case 'provider':
+            name = ep.name;
+            consumes = application;
+            scope = 'regular';
+            break;
+          case 'requirer':
+            provides = application;
+            break;
+        }
+      });
+      return (
+        <tr key={rel.id}>
+          <td>{name}</td>
+          <td>{provides}</td>
+          <td>{consumes}</td>
+          <td>{scope}</td>
+        </tr>
+      );
+    });
+    return (
+      <table key="relations">
+        <thead>
+          <tr>
+            <th>Relation</th>
+            <th>Provides</th>
+            <th>Consumes</th>
+            <th>Type</th>
+          </tr>
+        </thead>
+        <tbody>
+          {rows.sort(byKey)}
+        </tbody>
+      </table>
+    );
+  }
 
+};
 
 Status.propTypes = {
-  addNotification: PropTypes.func.isRequired
+  db: PropTypes.shape({
+    machines: PropTypes.object.isRequired,
+    relations: PropTypes.object.isRequired,
+    remoteServices: PropTypes.object.isRequired,
+    services: PropTypes.object.isRequired
+  }).isRequired,
+  model: PropTypes.shape({
+    cloud: PropTypes.string,
+    environmentName: PropTypes.string,
+    region: PropTypes.string,
+    sla: PropTypes.string,
+    version: PropTypes.string
+  }).isRequired,
+  urllib: PropTypes.func.isRequired
+};
+
+/**
+  Return an element class name suitable for the given value.
+  @params {String} value The provided value.
+  @returns {String} The class name ('ok', 'error' or '').
+*/
+const getClass = value => {
```

Comment:
They don't have to be methods, they don't need access to the class state. Is there a way to avoid possible name clashes while keeping them as simple functions?